app/shared/controllers/RequestHandler.scala

```diff
@@ -160,15 +163,21 @@ object RequestHandler {
             s"with correlationId : ${ctx.correlationId}")
 
         val result =
-          if (simulateRequestCannotBeFulfilled)
+          if (simulateRequestCannotBeFulfilled) {
             EitherT[Future, ErrorWrapper, Result](Future.successful(Left(ErrorWrapper(ctx.correlationId, RuleRequestCannotBeFulfilledError))))
-          else
+          } else {
             for {
               parsedRequest   <- EitherT.fromEither[Future](validator.validateAndWrapResult())
               serviceResponse <- EitherT(service(parsedRequest))
             } yield doWithContext(ctx.withCorrelationId(serviceResponse.correlationId)) { implicit ctx: RequestContext =>
-              handleSuccess(parsedRequest, serviceResponse)
+              responseModifier match {
+                case Some(responseHandler) =>
```

Comment:
could this just be "Some(modifier)"?  "responseHandler" is a bit confusing here alongside the RequestHandler etc